src/deploy/hosting/convertConfig.ts

```diff
@@ -57,27 +60,99 @@ export async function convertConfig(
     return out;
   }
 
-  const endpointBeingDeployed = (serviceId: string, region: string = "us-central1") => {
+  const endpointFromBackend = (
+    targetBackend: backend.Backend,
+    serviceId: string,
+    platform?: string,
+    region?: string
+  ): backend.Endpoint | undefined => {
+    const backendsForId = backend.allEndpoints(targetBackend).filter((endpoint) => {
+      return endpoint.id === serviceId;
+    });
+
+    const matchingBackends = backendsForId.filter((endpoint) => {
+      return (
+        (!region || endpoint.region === region) && (!platform || endpoint.platform === platform)
+      );
+    });
+
+    if (matchingBackends.length > 1) {
+      throw new FirebaseError(`More than one backend found for function name: ${serviceId}.
+      If the function is deployed in multiple regions, you must specify a region.`);
+    }
+
+    if (matchingBackends.length === 1) {
+      const endpoint = matchingBackends[0];
+      if (endpoint && isHttpsTriggered(endpoint)) {
+        return endpoint;
+      }
+    }
+    return undefined;
+  };
+
+  const endpointBeingDeployed = (
+    serviceId: string,
+    platform?: string,
+    region?: string
+  ): backend.Endpoint | undefined => {
     for (const { wantBackend } of Object.values(payload.functions || {})) {
-      const endpoint = wantBackend?.endpoints[region]?.[serviceId];
-      if (endpoint && isHttpsTriggered(endpoint) && endpoint.platform === "gcfv2") return endpoint;
+      if (!wantBackend) {
+        continue;
+      }
+      const endpoint = endpointFromBackend(wantBackend, serviceId, platform, region);
+      if (endpoint) {
+        return endpoint;
+      }
     }
     return undefined;
   };
 
-  const matchingEndpoint = async (serviceId: string, region: string = "us-central1") => {
-    const pendingEndpoint = endpointBeingDeployed(serviceId, region);
+  const matchingEndpoint = async (
+    serviceId: string,
+    platform?: string,
+    region?: string
+  ): Promise<backend.Endpoint | undefined> => {
+    const pendingEndpoint = endpointBeingDeployed(serviceId, platform, region);
     if (pendingEndpoint) return pendingEndpoint;
     const backend = await existingBackend(context);
     return allEndpoints(backend).find(
       (it) =>
         isHttpsTriggered(it) &&
-        it.platform === "gcfv2" &&
+        it.platform === platform &&
         it.id === serviceId &&
-        it.region === region
+        (region === undefined || it.region === region)
     );
   };
 
+  const findEndpointWithValidRegion = async (
+    rewrite: HostingRewrites,
+    context: Context
+  ): Promise<backend.Endpoint | undefined> => {
+    if ("function" in rewrite) {
```

Comment:
`in` can be fickle since it technically checks the object being looked at _and any object in its prototype chain_. That said, it's better to avoid it in my experience.

Would checking simply for `rewrite.function` be accurate? Or can it not be done for some reason?

Reply:
So I looked here:
https://www.typescriptlang.org/docs/handbook/advanced-types.html
https://levelup.gitconnected.com/user-defined-type-guards-in-typescript-fad639e4944f

And it seems like rewrite.function is not valid. The other ways is to use instanceof (which basically only works on classes) and a type predicate (which needs a named type, which we don't have). It seems like we have to leave it as is or redefine the type union and add some type predicates.